### PR TITLE
Update props table source error

### LINKED ISSUE #252: Better error message when props table can't find source.
# Feature Request

## Description
Props table can only be generated when a non transpiled file can be found. We automatically look for the non-transpiled file even if the consumer provides the lib file, but we fail to warn in a clear way.

We should warn with something like this:
`A non transpiled source file is required for the props table to be generated.`

## Additional Context / Screenshots
The error currently looks like this.
```
ERROR in ./node_modules/thing-js/lib/thing/components/ThingApplication.js?dev-site-props-table
Module build failed (from ./node_modules/terra-dev-site/config/webpack/loaders/devSitePropsTable.js):
Error: Can't resolve '/source/ui/node_modules/thing-js/src/thing/components/ThingApplication' in ''
    at doResolve (/source/ui/node_modules/enhanced-resolve/lib/Resolver.js:209:21)
    at hook.callAsync (/source/ui/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
    at _fn0 (eval at create (/source/ui/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at resolver.doResolve (/source/ui/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7)
    at hook.callAsync (/source/ui/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
    at _fn0 (eval at create (/source/ui/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at hook.callAsync (/source/ui/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
    at _fn1 (eval at create (/source/ui/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at hook.callAsync (/source/ui/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
    at _fn43 (eval at create (/source/ui/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at hook.callAsync (/source/ui/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
    at _fn0 (eval at create (/source/ui/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
    at resolver.doResolve (/source/ui/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43)
    at hook.callAsync (/source/ui/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
    at _fn2 (eval at create (/source/ui/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at hook.callAsync (/source/ui/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
 @ ./node_modules/thing-js/lib/terra-dev-site/doc/thing/Thing.1.doc.mdx 12:0-83 165:14-20
 @ ./dev-site-config/build/contentConfig.js
 @ ./dev-site-config/build/siteConfig.js
 @ ./node_modules/terra-dev-site/lib/TerraDevSite.js
```

### PR DESCRIPTION
**Pull request by DMcginn**
### Summary
Changed the error message in devSitePropsTable.js to be more informative as opposed to the previous code error block that was displayed when the props table could not find a source.

Closes #252 

### Testing
1. Compile the terra-dev-site project
2. Change the name of the [source file](https://github.com/cerner/terra-dev-site/blob/master/src/terra-dev-site/test/loaders/PropsTableExample.jsx), ex: `PropsTableExample.jsx` -> `PropsTableExample2.jsx`
3. Enter `npm run compile:prod` in the terminal
4. The updated error should appear as it does below and the project should be unable to compile.

```
ERROR in ./lib/terra-dev-site/test/loaders/PropsTableExample.js?dev-site-props-table
Module build failed (from ./config/webpack/loaders/devSitePropsTable.js):
Error: A non transpiled source file is required for the props table to be generated:
/Users/dm068655/Documents/terra-dev-site/src/terra-dev-site/test/loaders/PropsTableExample
```
Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by mjhenkes** [comment]
@DMcginn Could you add an example of the terminal output to the PR?

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,14 @@ Unreleased
 ### Fixed
 * Fixed failed to read sessionStorage Property error when the access is denied.
 
+### Changed
```
Comment (mjhenkes) [inline CHANGELOG.md:12]: Take a look at this change log, i think it got messed up during merging master in.

**Comment by DMcginn** [comment]
@mjhenkes @emilyrohrbough I added what the error currently looks like, I kept the previous error at the end of the block to provide more detail if it was needed. Additionally, the jest test was scrapped due to some difficulties.

File: config/webpack/loaders/devSitePropsTable.js [hunk 2]
```diff
@@ -125,7 +125,7 @@ const loader = async function loader(content) {
   // ensure src exists
   return this.resolve('', source, async (err, result) => {
     if (err) {
-      return callback(err);
+      return callback(new Error(`A non transpiled source file is required for the props table to be generated:\n${source}\n${err}`));
```
Comment (emilyrohrbough) [inline config/webpack/loaders/devSitePropsTable.js:128]: What is the error if we log like:
```suggestion
      return callback(new Error(`A non transpiled source file is required for the props table to be generated:\n${source}`));
```
To me, this is quite long and difficult to determine what went wrong

Comment (mjhenkes) [reply config/webpack/loaders/devSitePropsTable.js:128]: I agree, you kind of miss the first error.

Comment (DMcginn) [reply config/webpack/loaders/devSitePropsTable.js:128]: The would error look like this: 
```ERROR in ./lib/terra-dev-site/test/loaders/PropsTableExample.js?dev-site-props-table
Module build failed (from ./config/webpack/loaders/devSitePropsTable.js):
Error: A non transpiled source file is required for the props table to be generated:
/Users/dm068655/Documents/terra-dev-site/src/terra-dev-site/test/loaders/PropsTableExample
```
I agree the error being that large makes it hard to determine what is happening.

Comment (emilyrohrbough) [reply config/webpack/loaders/devSitePropsTable.js:128]: I think the smaller error would be easier to read & still provided info as to what file needs to be fixed to fix the error

**Pull request merged**